metta/rl/pufferlib/trainer.py

```
@@ -225,6 +227,25 @@ def __init__(
 
         logger.info(f"PufferTrainer initialization complete on device: {self.device}")
 
+        # W&B Stuck Run Detection - Initialize attributes
+        self.last_wandb_log_time = time.time()  # Start with current time
+        self._original_wandb_log_fn = None
+        self.wandb_stuck_monitor_thread = None
+
+        if self.wandb_run and self._master:
```

why use wandb instead of a tmpfile?

---

wandb.log calls are a good indication of whether a job is actually running or just in a silent error state